lib/teiserver_web/live/account/profile/overview.ex

```diff
@@ -362,18 +362,24 @@ defmodule TeiserverWeb.Account.ProfileLive.Overview do
   end
 
   def assign_accolade_notification(socket) do
-    user_id = socket.assigns.user.id
-    current_user_id = socket.assigns.current_user.id
-    days = 1
-
-    message =
-      with true <- user_id == current_user_id,
-           %{map: map, giver_name: giver_name} <- AccoladeLib.recent_accolade(user_id, days) do
-        "You have recently received an accolade from #{giver_name} for your match in #{map}!"
-      else
-        _ -> nil
-      end
+    message = get_accolade_notification_message(socket.assigns.user, socket.assigns.current_user)
 
     socket |> assign(:accolade_notification, message)
   end
+
+  def get_accolade_notification_message(viewed_user, current_user) do
+    days = 1
+
+    with true <- viewed_user != nil,
+         true <- current_user != nil,
+         true <- viewed_user.id == current_user.id && current_user.id != nil,
```

Comment:
You are checking if current user is not nil then on the next line additionally checking if current user's id isn't nil, Is this second id check necessary? If the current user is not nil why would its id be nil?

Reply:
The `with` construct is weird here as well, since you’re not using any pattern matching to bind any new variable. A regular `if` with a big `&&` is enough. `&&` is short circuiting anyway.